### ISSUE TITLE
Enable new image for Nouveau to be published in CI to Elastic Container Repository (ECR)

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-core#9685]
**Describe the issue**
A new service is being added to CHT Core, [Nouveau](https://github.com/medic/cht-core/issues/9542), and it needs a new image published via CI

**Describe the improvement you'd like**
Do any ECR and related work to allow the image to be published

**Comment by mrjones-plip** [comment]
@m5r reported running the CI job now, this error is returned:

`ERROR: failed to solve: failed to push 720541322708.dkr.ecr.eu-west-2.amazonaws.com/medic/cht-couchdb-nouveau:4.15.0-couchdb-nouveau.12124430292: unexpected status from HEAD request to https://720541322708.dkr.ecr.eu-west-2.amazonaws.com/v2/medic/cht-couchdb-nouveau/blobs/sha256:268db4a4b80fa4e519d1a26fa5d6382df8df33e19d63f21dc3bdb09b53dc56f2: 403 Forbidden`


@henokgetachew tried to add some perms in ECR:

```JSON
{
    "repository": {
        "repositoryArn": "arn:aws:ecr:eu-west-2:720541322708:repository/medic/cht-couchdb-nouveau",
        "registryId": "720541322708",
        "repositoryName": "medic/cht-couchdb-nouveau",
        "repositoryUri": "720541322708.dkr.ecr.eu-west-2.amazonaws.com/medic/cht-couchdb-nouveau",
        "createdAt": "2024-12-03T13:05:44.979000+03:00",
        "imageTagMutability": "MUTABLE",
        "imageScanningConfiguration": {
            "scanOnPush": false
        },
        "encryptionConfiguration": {
            "encryptionType": "AES256"
        }
    }
}
```

**Comment by mrjones-plip** [comment]
[5th try](https://github.com/medic/cht-core/actions/runs/12124430292/job/33862950048) is still failing the the `403` error [above](https://github.com/medic/cht-core/issues/9685#issuecomment-2515180097)

**Comment by henokgetachew** [comment]
The issue here was very simple that it was difficult to overlook. The repo i created was a private repo. Not a public one.

**Comment by henokgetachew** [comment]
@m5r can you now test?

**Comment by m5r** [comment]
@henokgetachew just gave it two more tries to be certain and it's still failing 😟

https://github.com/medic/cht-core/actions/runs/12273653842/job/34244889443?pr=9541#step:13:875

**Comment by henokgetachew** [comment]
Ah, now I see the issue! The way the builds work is that branch builds are pushed to the private repo whereas master builds are pushed to the public repo.

Private Registry Push:

`push 720541322708.dkr.ecr.eu-west-2.amazonaws.com/cht-couchdb-nouveau:<tag>`

Public Registry Push:

`push public.ecr.aws/medic/cht-couchdb-nouveau:<tag>`

The error we were seeing earlier was because the code is trying to push to the private registry but using the repository that was set up in the public registry.

Not sure where in the build that's configured (cc: @dianabarsan )

**Comment by jkuester** [comment]
@dianabarsan we are still facing this issue.  I have [a branch](https://github.com/medic/cht-core/pull/9541) building on to of the latest from `master` and it is failing with this same issue mentioned here (cannot push to private registry).  Could you help me figure out where we can update the config to be able to put  `cht-couchdb-nourveau` images to `720541322708.dkr.ecr.eu-west-2.amazonaws.com`?  🙏  Thanks!

**Comment by mrjones-plip** [comment]
Here's the full error in case it gets lost/is easier to read inline (see below).  I'll take a look around and compare the ECR settings to other repos in case I see anything that stands out.

I can just "re-run failed jobs" to test, yeah @jkuester ?

```
 #10 exporting manifest list sha256:8e61563c2aa2416a836aac3b405ccec6f9a561955701daed328aa4efb0205c5a done
#10 pushing layers
#10 pushing layers 0.2s done
#10 ERROR: failed to push 720541322708.dkr.ecr.eu-west-2.amazonaws.com/medic/cht-couchdb-nouveau:4.17.0-couchdb-nouveau.13465388266: unexpected status from HEAD request to https://720541322708.dkr.ecr.eu-west-2.amazonaws.com/v2/medic/cht-couchdb-nouveau/blobs/sha256:248373f431090f8544b7cf9f3f3ec53d16c57e344c2f86332631c2eb9db21efc: 403 Forbidden

#11 [auth] sharing credentials for 720541322708.dkr.ecr.eu-west-2.amazonaws.com
#11 DONE 0.0s
------
 > exporting to image:
------
ERROR: failed to solve: failed to push 720541322708.dkr.ecr.eu-west-2.amazonaws.com/medic/cht-couchdb-nouveau:4.17.0-couchdb-nouveau.13465388266: unexpected status from HEAD request to https://720541322708.dkr.ecr.eu-west-2.amazonaws.com/v2/medic/cht-couchdb-nouveau/blobs/sha256:248373f431090f8544b7cf9f3f3ec53d16c57e344c2f86332631c2eb9db21efc: 403 Forbidden
/home/runner/work/cht-core/cht-core/scripts/build/index.js:223
      return reject(new Error(`${command} exited with ${code}`));
                    ^

Error: docker exited with 1
    at ChildProcess.<anonymous> (/home/runner/work/cht-core/cht-core/scripts/build/index.js:223:21)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1104:16)
    at ChildProcess._handle.onexit (node:internal/child_process:304:5)

Node.js v22.11.0
Error: Process completed with exit code 1.
```

**Comment by jkuester** [comment]
> I can just "re-run failed jobs" to test, yeah @jkuester ?

Yes! (It will take a bit to get through the whole build/unit-test process, but then it should re-trigger the image publish.)

**Comment by mrjones-plip** [comment]
hmm - i was hoping the ECR permission settings would have a meaningful difference - I do see a few, but none that look meaningful.  I'll try comparing CI YAML 🤷 

![Image](https://github.com/user-attachments/assets/0286140d-3bf4-4022-84fc-88ca02fdba50)

**Comment by mrjones-plip** [comment]
floundered around for a bit, jumping between the [YAML](https://github.com/medic/cht-core/blob/d5499ae619f74869f10fee8c9e22667b77837aab/.github/workflows/build.yml#L420) and the [JS](https://github.com/medic/cht-core/blob/d5499ae619f74869f10fee8c9e22667b77837aab/scripts/build/index.js#L241), but didn't find anything.  I suspect we'd either need to trace this through to add debug steps or, yeah, reach out to Diana as you did already already has all this uploaded to wet ware 🧠 

sorry I can't assist!

**Comment by dianabarsan** [comment]
@jkuester and @mrjones-plip 
I have no idea about what to change about this repository, I think I don't have the permissions to do so and can't find anything in the UI. 
Requesting @Hareet to detangle us, please!

**Comment by Hareet** [comment]
@mrjones-plip @dianabarsan 
Completed. Please test and let me know if its not working. 

There's an IAM Group policy called `ecr-publishers` that is tied to IAM user `ci-user`.
Inside `ecr-publishers` policy, you have to add a line for the resource you create in ecr.

<img width="926" alt="Image" src="https://github.com/user-attachments/assets/e5deb42b-9fc6-4eb2-b90c-9594df0356d8" />

**Comment by mrjones-plip** [comment]
Thanks @Hareet !  I kicked off [this action](https://github.com/medic/cht-core/actions/runs/13465388266) which I think is the right one to test publishing the image?  @jkuester will sort me out if I fired off the wrong one 😅

**Comment by dianabarsan** [comment]
Thanks @Hareet !!

**Comment by jkuester** [comment]
We are pushing images!!!!  Thank you everyone for your support here! 🚀
